packages/cached/src/cached-fs.ts

```diff
@@ -55,6 +67,31 @@ export function createCachedFs(fs: IFileSystem): ICachedFileSystem {
         invalidateAbsolute(directoryPath);
         return fs.mkdirSync(directoryPath);
       },
+      readFile: function readFile(path: string, callback: CallbackFn<string | Buffer>) {
+        path = fs.resolve(path);
+        const cacheKey = getCanonicalPath(path);
```

Comment:
this is not enough for a readfile cache key.
`readFile` has optional options that affects the return type. "encoding" specifically. It gets a bit tricky to represent it in the callback version, I know.

add a test that:
```ts
const fileBuffer = readFile('/some/file/path')
const fileAsString = readFile('/some/file/path', {encoding: 'utf8'})
```
and ensure they are not the same.

and now's the catch. `@file-services/memory` never retuns a real Node `Buffer`. it returns the `string`, which conforms to the *partial* Buffer interface exposed (only `toString()` method). You'll need to test this specific instance with `@file-services/node`. Node-only test.